msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -82,30 +97,44 @@ public abstract partial class ClientApplicationBase
+        /// The return value of this propety is either the value provided by the developer in the constructor of the application, or otherwise 
+        /// the value of the <see cref="DefaultAuthority"/> static member (that is <c>https://login.microsoftonline.com/common/</c>)
         /// </Summary>
         public string Authority { get; }
 
         /// <summary>
-        /// Will be a default value. Can be overridden by the developer. Once set, application will bind to the client Id.
+        /// Gets the Client ID (also named Application ID) of the application as registered in the application registration portal (https://aka.ms/msal-net-register-app)
+        /// and as passed in the constructor of the application
         /// </summary>
         public string ClientId { get; }
 
         /// <summary>
-        /// Redirect Uri configured in the app registration portal. PublicClientApplication has a default value of 
-        /// urn:ietf:wg:oauth:2.0:oob.This default does not apply to iOS and Android as the library needs to leverage 
-        /// system webview for authentication.
+        /// The redirect URI (also named Reply URI), is the URI at which Azure AD will contact back the application with the tokens. 
+        /// This redirect URI needs to be registered in the app registration (https://aka.ms/msal-net-register-app)
+        /// In MSAL.NET, <see cref="T:PublicClientApplication"/> define the following default RedirectUri values:
+        /// <list type="bullet">
+        /// <item><c>urn:ietf:wg:oauth:2.0:oob</c> for desktop (.NET Framework and .NET Core) applications</item>
+        /// <item><c>msal{ClientId}</c> for Xamarin iOS and Xamarin Android (as this will be used by the system web browser by default on these
+        /// platforms to call back the application)
+        /// </item>
+        /// </list>
+        /// In <see cref="T:ConfidentialClientApplication"/>, this can be the URL of the Web application / Web API.
         /// </summary>
+        /// <remarks>This is especially important when you deploy an application that you have initially tested locally; 
+        /// you then need to add the reply URL of the deployed application in the application registration portal</remarks>
         public string RedirectUri { get; set; }
 
         /// <summary>
-        /// Sets or Gets the custom query parameters that may be sent to the STS for dogfood testing. This parameter should not be set by the 
-        /// developers as it may have adverse effect on the application.
+        /// Sets or Gets a custom query parameters that may be sent to the STS for dogfood testing or debugging. This is a string of segments
+        /// of the form <c>key=value</c> separated by an ampersand character.
+        /// Unless requested otherwise, this parameter should not be set by application developers as it may have adverse effect on the application.
         /// </summary>
         public string SliceParameters { get; set; }
```

Comment:
Can this be used in the same way as extraqueryparameters? It seems like it.

Reply:
it is. added a sentence.